Keen.NetStandard.Test/DatasetTests_Integration.cs

```diff
@@ -0,0 +1,434 @@
+        {
+            var apiResponse = File.ReadAllText($"{GetApiResponsesPath()}/ListDatasetDefinitions.json");
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForGetAsync(apiResponse);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+            var datasetCollection = client.ListDatasetDefinitions();
+
+            Assert.IsNotNull(datasetCollection);
+            Assert.IsNotNull(datasetCollection.Datasets);
+            Assert.IsTrue(datasetCollection.Datasets.Any());
+            Assert.IsTrue(!string.IsNullOrWhiteSpace(datasetCollection.NextPageUrl));
+
+            foreach (var item in datasetCollection.Datasets)
+            {
+                AssertDatasetIsPopulated(item);
+            }
+        }
+
+        [Test]
+        public void ListAllDefinitions_Success()
+        {
+            var apiResponse = File.ReadAllText($"{GetApiResponsesPath()}/ListDatasetDefinitions.json");
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForGetAsync(apiResponse);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+            var datasetCollection = client.ListAllDatasetDefinitions();
+
+            Assert.IsNotNull(datasetCollection);
+            Assert.IsTrue(datasetCollection.Any());
+
+            foreach (var item in datasetCollection)
+            {
+                AssertDatasetIsPopulated(item);
+            }
+        }
+
+        [Test]
+        public void Delete_Success()
+        {
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForDeleteAsync(string.Empty);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+
+            client.DeleteDataset("datasetName");
+        }
+
+        [Test]
+        public void CreateDataset_Success()
+        {
+            var apiResponse = File.ReadAllText($"{GetApiResponsesPath()}/GetDatasetDefinition.json");
+
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForPutAsync(apiResponse);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+            var newDataSet = CreateDatasetDefinition();
+            var dataset = client.CreateDataset(newDataSet);
+
+            AssertDatasetIsPopulated(dataset);
+        }
+
+        [Test]
+        public void DatasetValidation_Throws()
+        {
+            var dataset = new DatasetDefinition();
+
+            Assert.Throws<KeenException>(() => dataset.Validate());
```

Comment:
This test seems like it doesn't quite cover all validation. For example it doesn't really test that AnalysisType being unset will ever throw. Maybe separate tests for each invalid condition would be warranted? Maybe start with a valid dataset and then modify only one field at a time per-test and assert that it throws?

Reply:
Added a blurb to Issue #130 about this.